The entity name must immediately follow the '&' in the entity reference

- Plug-in Version 3.4.0

As for the title the error message it's the same 

> "The entity name must immediately follow the '&' in the entity reference"

tested also on version 3.3.1 and this version doesn't have this issue.

Thanks

This started happening in our project when we updated:
from AGP 4.2.2 + gradle 6.9 + java 8
to AGP 8.3.0 + gradle 8.4 + java 17 (build), java 11 (source)

It happens both with plugin version 3.4.2 and 4.2.0.

```
Caused by: org.xml.sax.SAXParseException; lineNumber: 1601; columnNumber: 59; The entity name must immediately follow the '&' in the entity reference.
        at java.xml/com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:262)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:342)
        at com.hyperdevs.poeditor.gradle.ktx.DocumentExtensionsKt.toStringsXmlDocument(DocumentExtensions.kt:41)
        at com.hyperdevs.poeditor.gradle.xml.XmlPostProcessor.splitTranslationXml(XmlPostProcessor.kt:113)
        at com.hyperdevs.poeditor.gradle.xml.XmlPostProcessor.postProcessTranslationXml(XmlPostProcessor.kt:55)
        at com.hyperdevs.poeditor.gradle.PoEditorStringsImporter.importPoEditorStrings(PoEditorStringsImporter.kt:146)
        at com.hyperdevs.poeditor.gradle.tasks.ImportPoEditorStringsTask.importPoEditorStrings(ImportPoEditorStringsTask.kt:184)

```

Nice that `unescapeHtmlTags = false` fixes the issue. It introduces some other changes though. Some seem fine, so can be problematic.

This seems fine - it changes emoji codes/formats, but they still work. 
For instance, it changes this string `&#x1f929;` into `&#129321;`. Both resolve to the same emoji - 🤩. The first one is the emoji Hex code point (as it was originally, before upgrading AGP, gradle and Java). The second one is the Decimal code point of the same emoji.

**This does not seem fine though** - it encodes tags, so they don't work anymore. For instance, it breaks tags like <b>. 
You might be using this tag to make text bold in the Android UI (works even in Toast messages). For instance: `John is \"<b>awesome</b>\", really!` to output "John is "**awesome**", really!", it will be broken when using `unescapeHtmlTags = false`: `John is \"&lt;b&gt;awesome&lt;/b&gt;\", really!` - which appears exactly like that in the UI.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -48,6 +48,7 @@ poEditor {
 ```
 
 </details>
+
 ### Changed
 - Refactor the whole parsing logic to better handle input and improve performance.
 ### Deprecated
@@ -55,7 +56,7 @@ poEditor {
 ### Removed
 - No removed features!
 ### Fixed
-- No fixed issues!
+- Fix bug with `&` character not being properly escaped.
 ### Security
 - No security issues fixed!
 

--- a/README.md
+++ b/README.md
@@ -123,7 +123,7 @@ Attribute                              | Description
 ```resFileName```                      | (Since 3.1.0) (Optional) Sets the file name for the imported string resource XML files. Defaults to `strings`.
 ```order```                            | (Since 3.1.0) (Optional) Defines how to order the export. Accepted values are defined by the POEditor API.
 ```unquoted```                         | (Since 3.2.0) (Optional) Defines if the strings should be unquoted, overriding default PoEditor configuration. Defaults to `false`.
-```unescapeHtmlTags```                 | (Since 3.4.0) (Optional) Whether or not to unescape HTML entitites from strings. Defaults to `true`.
+```unescapeHtmlTags```                 | (Since 3.4.0) (Optional) Whether or not to unescape HTML tags (`<`, `>`) from strings. Defaults to `true`.
 ```untranslatableStringsRegex```       | (Since 4.2.0) (Optional) Pattern to use to mark strings as translatable=false in the strings file. Defaults to `null`.
 ```includeComments```                  | (Since 5.0.0) (Optional) Whether to include comments from the downloaded strings. Defaults to `true`.
 

--- a/src/main/kotlin/com/hyperdevs/poeditor/gradle/extensions/StringExtensions.kt
+++ b/src/main/kotlin/com/hyperdevs/poeditor/gradle/extensions/StringExtensions.kt
@@ -26,9 +26,6 @@ private val cdataRegex = Regex("^<!\\[CDATA\\[(.*)]]>$", RegexOption.DOT_MATCHES
 fun String.unescapeHtmlTags() = this
     .replace("&lt;", "<")
     .replace("&gt;", ">")
-    .replace("&amp;", "&")
-    .replace("&apos;", "'")
-    .replace("&quot;", "\"")
 
 /**
  * Returns if the given string is a CDATA string.

--- a/src/test/kotlin/com/hyperdevs/poeditor/gradle/xml/StringsXmlPostProcessorTest.kt
+++ b/src/test/kotlin/com/hyperdevs/poeditor/gradle/xml/StringsXmlPostProcessorTest.kt
@@ -432,24 +432,6 @@ class StringsXmlPostProcessorTest {
     }
 
     @Test
-    fun `Postprocessing XML with string HTML symbols works 2`() {
-        // Test complete Xml
-        val inputStringsXmlDocument = StringsXmlDocument(
-            resources = listOf(
-                StringsXmlResource.StringElement("hello_friend_bold", "&amp;lt;b&amp;gt;Hello world&amp;lt;/b&amp;gt;")
-            )
-        )
-
-        val expectedResult = StringsXmlDocument(
-            resources = listOf(
-                StringsXmlResource.StringElement("hello_friend_bold", "&lt;b&gt;Hello world&lt;/b&gt;")
-            )
-        )
-
-        assertEquals(expectedResult, StringsXmlPostProcessor.formatTranslationXml(inputStringsXmlDocument, true, null, true))
-    }
-
-    @Test
     fun `Postprocessing XML with string HTML symbols and unescape set to false works`() {
         // Test complete Xml
         val inputStringsXmlDocument = StringsXmlDocument(

